frame/evm/src/runner/stack.rs

```diff
@@ -949,221 +892,29 @@ where
-					let maybe_record = !recorded.account_codes.contains(&address);
-					// Skip if the address has been already recorded this block
-					if maybe_record {
-						// First we record account emptiness check.
-						// Transfers to EOAs with standard 21_000 gas limit are able to
-						// pay for this pov size.
-						weight_info.try_record_proof_size_or_fail(IS_EMPTY_CHECK_PROOF_SIZE)?;
-						if <AccountCodes<T>>::decode_len(address).unwrap_or(0) == 0 {
-							return Ok(());
-						}
-
-						weight_info
-							.try_record_proof_size_or_fail(ACCOUNT_CODES_METADATA_PROOF_SIZE)?;
-						if let Some(meta) = <AccountCodesMetadata<T>>::get(address) {
-							weight_info.try_record_proof_size_or_fail(meta.size)?;
-						} else if let Some(remaining_proof_size) =
-							weight_info.remaining_proof_size()
-						{
-							let pre_size = remaining_proof_size.min(size_limit);
-							weight_info.try_record_proof_size_or_fail(pre_size)?;
-
-							let actual_size = Pallet::<T>::account_code_metadata(address).size;
-							if actual_size > pre_size {
-								return Err(ExitError::OutOfGas);
-							}
-							// Refund unused proof size
-							weight_info.refund_proof_size(pre_size.saturating_sub(actual_size));
-						}
-						recorded.account_codes.push(address);
-					}
-				}
-				ExternalOperation::IsEmpty => {
-					weight_info.try_record_proof_size_or_fail(IS_EMPTY_CHECK_PROOF_SIZE)?
-				}
-				ExternalOperation::Write(_) => {
-					weight_info.try_record_proof_size_or_fail(WRITE_PROOF_SIZE)?
-				}
-			};
-		}
+	fn record_external_operation(&mut self, _op: evm::ExternalOperation) -> Result<(), ExitError> {
 		Ok(())
 	}
 
 	fn record_external_dynamic_opcode_cost(
 		&mut self,
-		opcode: Opcode,
+		_opcode: Opcode,
 		_gas_cost: GasCost,
-		target: evm::gasometer::StorageTarget,
+		_target: evm::gasometer::StorageTarget,
 	) -> Result<(), ExitError> {
-		// If account code or storage slot is in the overlay it is already accounted for and early exit
-		let accessed_storage: Option<AccessedStorage> = match target {
-			StorageTarget::Address(address) => {
-				if self.recorded().account_codes.contains(&address) {
-					return Ok(());
-				} else {
-					Some(AccessedStorage::AccountCodes(address))
-				}
-			}
-			StorageTarget::Slot(address, index) => {
-				if self
-					.recorded()
-					.account_storages
-					.contains_key(&(address, index))
-				{
-					return Ok(());
-				} else {
-					Some(AccessedStorage::AccountStorages((address, index)))
-				}
-			}
-			_ => None,
-		};
-
-		let size_limit: u64 = self
-			.metadata()
-			.gasometer()
-			.config()
-			.create_contract_limit
-			.unwrap_or_default() as u64;
-		let (weight_info, recorded) = self.info_mut();
-
-		if let Some(weight_info) = weight_info {
-			// proof_size_limit is None indicates no need to record proof size, return directly.
-			if weight_info.proof_size_limit.is_none() {
-				return Ok(());
-			};
-
-			let mut record_account_codes_proof_size =
```

Comment:
We should not remove theses proof size check, otherwise the evm execution can exceed proof size limit, and we can't revert storage reads, so we must keep all proof size checks even if we use the proof_size host function to reclaim the diff at the end

Reply:
Hmm, if we are going to add those hooks back,  seems there is no need to introduce the host function. The rational behind this pr is trying to avoid using the evm hook and keep the evm clean.

Reply:
I think the host function is still useful to refund extra proof size, because the "computed proof size" overestimate a bit (we have to overestimate to account for the worst case scenario)